compiler/test/runnable/memberOfProperty.d

```diff
@@ -0,0 +1,41 @@
+extern(C) void printf(const char*, ...);
+
+extern(C) void main() {
+    int varB = :max;
+    Context varS = :max;
+    Enum varE = :End;
+    Enum varF1 = functionReturn();
+    Enum varF2 = identity(:Middle);
+    printf("%d <> %d <> %d <> %d <> %d\n", varB, varS.value, varE, varF1, varF2);
+
+    switch(varF1) {
+        case :Start:
+            static assert(!__traits(compiles, { const fail = :max; }));
+            break;
+        default:
+            assert(0);
+    }
+
+    static assert(!__traits(compiles, { const fail = :max; }));
+    static assert(!__traits(compiles, { Context fail = :max.min; }));
```

Comment:
silly question: why does this fail? this breaks my metal model of `:` as shorthand for `typeof(lhs)` (or `typeof(return)` as appropriate).

Reply:
Sounds like it might be more useful to have a token which represents the expected type, so you can call its constructor:
`fun(_(5), _.some_member)`.